src/Stan/Report/Settings.hs

```diff
@@ -8,13 +8,40 @@ Report settings types.
 
 module Stan.Report.Settings
     ( ReportSettings (..)
-    ) where
 
-import Stan.Core.Toggle (ToggleSolution)
+      -- * Verbosity
+    , Verbosity (..)
+    , isVerbose
+      -- * Toggle
+    , ToggleSolution (..)
+    , isHidden
+    ) where
 
 
 {- | Settings for produced report.
 -}
 data ReportSettings = ReportSettings
-    { reportSettingsSolutionVerbosity :: !ToggleSolution
+    { reportSettingsVerbosity         :: !Verbosity
+    , reportSettingsSolutionVerbosity :: !ToggleSolution
     }
+
+data Verbosity
+    = Verbose
+    | NonVerbose
+    deriving stock (Show)
+
+isVerbose :: Verbosity -> Bool
+isVerbose = \case
+     Verbose -> True
+     NonVerbose -> False
+
+-- | Boolean for showing/hiding solution information of observations.
+data ToggleSolution
```

Comment:
@chshersh , do you think that I should use verbosity instead of toggle here?

Reply:
@vrom911 No, I don't think so. I think it's okay to have different `Bool`-like enums to make code more readable.